cpp/sopt/mpi/communicator.h

```diff
@@ -446,7 +446,7 @@ CONTAINER Communicator::gather_(CONTAINER const &vec, std::vector<t_int> const &
   assert(root < size());
   if (sizes.size() != size() and rank() == root)
     throw std::runtime_error("Sizes and communicator size do not match on root");
-  else if (rank() != root and sizes.size() != 0 and sizes.size() != size())
```

Comment:
Somehow my brain likes replacing `size == 0` with `empty()` but prefers `size != 0` over `!empty()`. I think the `!` on `!empty()` is quite easy to not notice so if I was reading the code quickly I might get the logic wrong.

Is this just me? @tkoskela @mmcleod89 what do you think?

Reply:
Personally I do use `!empty()` quite a lot and have seen it quite a bit in production code; I have generally found it quite easy to read because I find `!` always jumps out at me!